### DISCUSSION TITLE
Click and Double Click

### DISCUSSION THREAD

**Discussion by BctfN0HUK7Yg** [emilk/egui#1107] [Q&A]
I have response.clicked() and response.double_clicked() with a different actions. 
But double clicked never happend becouse single clicked act and change my ui.
How to react on Double click? And how Double click can happend within frame?

**Comment by BctfN0HUK7Yg** [comment]
I found solution with counting time after click and when nothing happend react for click. But I have to use request_repaint to update input().time

**Reply by emilk** [reply]
That solution adds input delay in case of clicks.

In my opinion, a better UX is that clicks happen directly. So for instance, if a single click selects and a double click opens, then double-clicking anything will select it first, then open it. And that's fine.
